manifests/webhook/package.pp

```diff
@@ -1,19 +1,13 @@
 # Private class, do not include it directly.
 # Installs the webhook packages
 class r10k::webhook::package (
-  $is_pe_server = $r10k::params::is_pe_server,
-  $provider     = $r10k::params::provider,
+  $is_pe_server    = $r10k::params::is_pe_server,
+  $provider        = $r10k::params::provider,
+  $sinatra_version = $r10k::params::webhook_sinatra_version,
 ) inherits r10k::params {
   if !defined(Package['sinatra']) {
     package { 'sinatra':
-      ensure   => installed,
-      provider => $provider,
-      before   => Service['webhook'],
-    }
-  }
-  if !defined(Package['rack']) {
-    package { 'rack':
-      ensure   => installed,
+      ensure   => $sinatra_version,
```

Comment:
Right, but `installed` works with all providers, whereas `~> 1.0` doesn't work with many of them. This change could be a breaking change to someone using a non-standard provider, so I think it may be helpful to promote it in the README. I'll look tomorrow to see if there's an appropriate place unless you find one first.